lib/scala/project-manager/src/test/scala/org/enso/projectmanager/protocol/MissingComponentBehavior.scala

```diff
@@ -56,4 +57,26 @@ trait MissingComponentBehavior {
       client.expectTaskStarted()
     }
   }
+
+  /** This behaviour should be tested in a separate test suite, as it affects
+   * the test environment and if run together with other tests it could affect
+   * their results.
+   */
+  def correctlyHandleMissingRuntimeInPresenceOfEngine(): Unit = {
+    "make sure to check if the runtime is installed even if the engine was " +
+      "already installed" in {
+      uninstallRuntime(GraalVMVersion("23.2.0", "21.0.0"))
+
+      val client = new WsTestClient(address)
+      client.send(
+        buildRequest(defaultVersion, MissingComponentAction.Install)
+      )
+
+      /** We do not check for success here as we are concerned onyl that the
+       * installation is attempted. Installation and creating/opening projects
```

Comment:
Because the installation was explicitly requested by the client at line 72 [here](https://github.com/enso-org/enso/blob/851fabf06fafbb989e92c62f7ae73b75ab99c765/lib/scala/project-manager/src/test/scala/org/enso/projectmanager/protocol/MissingComponentBehavior.scala#L72).